homeassistant/components/mqtt/binary_sensor.py

```diff
@@ -114,6 +117,43 @@ def __init__(self, hass, config, config_entry, discovery_data):
 
         MqttEntity.__init__(self, hass, config, config_entry, discovery_data)
 
+    async def async_added_to_hass(self) -> None:
+        """Restore state for entities with expire_after set."""
+        await super().async_added_to_hass()
+        if (
+            (expire_after := self._config.get(CONF_EXPIRE_AFTER)) is not None
+            and expire_after > 0
+            and (last_state := await self.async_get_last_state()) is not None
+            and last_state.state not in [STATE_UNKNOWN, STATE_UNAVAILABLE]
+        ):
+            expiration_at = last_state.last_changed + timedelta(seconds=expire_after)
+            if expiration_at < (time_now := dt_util.utcnow() + timedelta(seconds=5)):
```

Comment:
Why do we not start a timer if the remaining time is less than 5 seconds, can't we just check if it's less than 0?

Reply:
It will take some processing time to setup the timer. but less than 0 can be done too.